projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -251,80 +235,51 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
    * @param InternalLibImage[] images is the array of all images.
    */
   private initPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
-    let index: number;
```

Comment:
Same here: for simplification and edge-case handling, I propose that the whole logic be processed uniformly in a single method (setIndexesPreviews)